### ISSUE TITLE
Swipeout to delete lists

### ISSUE THREAD

**Issue by asantos00** [bencompton/framework7-react#42]
Hi!

I'm having this problem:

I use swipeout to delete lists, it feels very native and it is awesome, but I'm having a problem, I use the `onSwipeoutDeleted` callback to make a request to the server and delete that item, and I'm currently using meteor. When I delete that on server, my view re-renders, and tries to remove the list item that is not there cause (I think) it is "unsafely removed" from DOM, without react noticing.

Is there an option to use so framework7 don't really remove the element? Am I doing something wrong?

Thanks and keep um the good work!

**Comment by bencompton** [comment]
Yeah, having Framework7 JavaScript remove DOM elements outside of React isn't the best idea. The way it currently works in Framework7 is that the "deleted" event gets fired after the CSS animation completes and then it deletes it. There currently is no option to prevent it from deleting the list item from the DOM after the animation. Ideally, there would be such an option, and then it would be possible to have React re-render and delete it after the animated transition occurs. It would take a PR to Framework7 to accomplish that. If you look in this code, you can see what I'm talking about: https://github.com/nolimits4web/Framework7/blob/master/src/js/framework7/swipeout.js#L394

If you can live without the nice delete animation, you could add a  `ListItemSwipeoutButton` without the `delete` prop and an `onClick` handler. You could then trigger the React component to re-render and remove that item. You would probably want to immediately do an optimistic update to your client-side state so the list item disappears right away, and then shoot off the request to the server to actually delete it for real. That's how I would do it with Redux anyhow, but I haven't really worked with Meteor.
